packages/kit/src/core/build/index.js

```diff
@@ -223,179 +265,56 @@ async function build_server(
 		fs.writeFileSync(hooks_file, '');
 	}
 
-	const app_file = `${build_dir}/app.js`;
+	const app_file = `${build_dir}/app.js`; // TODO rename
 
-	/** @type {(file: string) => string} */
-	const app_relative = (file) => {
-		const relative_file = path.relative(build_dir, path.resolve(cwd, file));
-		return relative_file[0] === '.' ? relative_file : `./${relative_file}`;
+	/** @type {Record<string, string>} */
+	const input = {
+		// TODO
+		app: app_file
 	};
 
-	const prefix = `/${config.kit.appDir}/`;
-
-	/**
-	 * @param {string} file
-	 * @param {Set<string>} js_deps
-	 * @param {Set<string>} css_deps
-	 */
-	function find_deps(file, js_deps, css_deps) {
-		const chunk = client_manifest[file];
-
-		if (js_deps.has(chunk.file)) return;
-		js_deps.add(chunk.file);
-
-		if (chunk.css) {
-			chunk.css.forEach((file) => css_deps.add(file));
-		}
-
-		if (chunk.imports) {
-			chunk.imports.forEach((file) => find_deps(file, js_deps, css_deps));
+	manifest_data.routes.forEach((route) => {
+		if (route.type === 'endpoint') {
+			const resolved = path.resolve(cwd, route.file);
+			const relative = path.relative(config.kit.files.routes, resolved);
+			const name = posixify(path.join('entries/endpoints', relative.replace(/\.js$/, '')));
+			input[name] = resolved;
 		}
-	}
-
-	/** @type {Record<string, { entry: string, css: string[], js: string[], styles: string[] }>} */
-	const metadata_lookup = {};
-
-	manifest.components.forEach((file) => {
-		const js_deps = new Set();
-		const css_deps = new Set();
-
-		find_deps(file, js_deps, css_deps);
-
-		const js = Array.from(js_deps);
-		const css = Array.from(css_deps);
+	});
 
-		const styles = config.kit.amp
-			? Array.from(css_deps).map((url) => {
-					const resolved = `${output_dir}/client/${config.kit.appDir}/${url}`;
-					return fs.readFileSync(resolved, 'utf-8');
-			  })
-			: [];
+	manifest_data.components.forEach((file) => {
```

Comment:
something that could use a comment: why a separate loop here over `manifest_data.components`? i.e. why does the loop above only handle endpoints?

Reply:
endpoints are defined by a single module, but pages are defined by multiple components (plan a: [layout, layout, page], plan b: [error]). since there's component overlap between pages (many pages share layout/error components) we loop over the deduped list instead. added a comment